hudi-platform-service/hudi-metaserver/src/main/java/org/apache/hudi/metaserver/client/HoodieMetaserverClientImp.java

```diff
@@ -108,35 +108,42 @@ public void createTable(Table table) {
     }
   }
 
+  @Override
   public List<HoodieInstant> listInstants(String db, String tb, int commitNum) {
     return exceptionWrapper(() -> this.client.listInstants(db, tb, commitNum).stream()
-        .map(EntityConvertor::fromTHoodieInstant)
+        .map(EntityConversions::fromTHoodieInstant)
         .collect(Collectors.toList())).get();
   }
 
+  @Override
   public Option<byte[]> getInstantMeta(String db, String tb, HoodieInstant instant) {
-    ByteBuffer bytes = exceptionWrapper(() -> this.client.getInstantMeta(db, tb, EntityConvertor.toTHoodieInstant(instant))).get();
-    Option<byte[]> res = bytes.capacity() == 0 ? Option.empty() : Option.of(bytes.array());
-    return res;
+    ByteBuffer byteBuffer = exceptionWrapper(() -> this.client.getInstantMeta(db, tb, EntityConversions.toTHoodieInstant(instant))).get();
+    byte[] bytes = new byte[byteBuffer.remaining()];
+    byteBuffer.get(bytes);
+    return byteBuffer.hasRemaining() ? Option.empty() : Option.of(bytes);
```

Comment:
my bad. this condition should be flipped. would also change this to just return `byte[]` if you agree

Reply:
Flink will commit the instant with empty instant meta? I try to find the code to prove it.

Reply:
> Flink will commit the instant with empty instant meta?

sounds like it's a responsibility of who commits the instant to decide what to write. Here we better stick to what is desirable from API design perspective. Also would like to see the relevant code to make a better judgement

Reply:
There is the basic api which provides commit an empty instant meta
![image](https://user-images.githubusercontent.com/17903481/210355049-cc304ae1-495c-4c8a-b9ef-6228204aa0a6.png)